sdk/trace/src/main/java/io/opentelemetry/sdk/trace/SpanProcessor.java

```diff
@@ -99,4 +101,13 @@ default CompletableResultCode shutdown() {
   default CompletableResultCode forceFlush() {
     return CompletableResultCode.ofSuccess();
   }
+
+  /**
+   * Closes this {@link SpanProcessor} after processing any remaining spans, releasing any
+   * resources.
+   */
+  @Override
+  default void close() {
+    shutdown().join(10, TimeUnit.SECONDS);
```

Comment:
This seems a bit questionable. If we have a shutdown timeout, should it be hardcoded or configurable? If it is hardcoded, maybe use a constant?

Reply:
Also, maybe it would be better to throw a timeout exception if the shutdown does not complete.

Reply:
I think we expect this to finish relatively quickly, just finish things up. So maybe 10 seconds was too large, let me try 1 second. I'd prefer hard-coded just to keep things simpler, if extracting a constant are you thinking `internal` package?

> Also, maybe it would be better to throw a timeout exception if the shutdown does not complete.

I don't know if I've ever been able to handle an exception during shutdown so am ambivalent, but can add it. How useful does it seem?

Reply:
> I think we expect this to finish relatively quickly

Hmm, since shutdown includes flush, it will often make network requests.